presto-main/src/main/java/com/facebook/presto/execution/NodeTaskMap.java

```diff
@@ -129,9 +129,11 @@ public NodeStatsTracker createTaskStatsTrackers(TaskId taskId)
 
             // when nodeStatsTracker is garbage collected, run the cleanup method on the tracker
             // Note: tracker can not have a reference to nodeStatsTracker
-            finalizerService.addFinalizer(nodeStatsTracker, splitTracker::cleanup);
-            finalizerService.addFinalizer(memoryUsageTracker, memoryUsageTracker::cleanup);
-            finalizerService.addFinalizer(cpuUtilizationPercentageTracker, cpuUtilizationPercentageTracker::cleanup);
+            finalizerService.addFinalizer(nodeStatsTracker, () -> {
```

Comment:
perhaps, add a comment explaining that this cannot be broken down into 3 separate finalizers as this is non-obvious

Reply:
I have added the necessary comment